src/2_intro.md

```diff
@@ -1,20 +1,28 @@
 # Introduction to Machine Learning with Rust
 
-TO DO:
-- Description of the Rust-ML group
-- Introduction to the ecosystem
-    - ndarray common linear algebra backend for scientific computing
-    - linfa is the Rust-ML group's meta-crate for classical machine learning and data science, analogous to Python's scikit-learn
-    - The data visualization is generally provided by the `plotters` crate
-- Reminder about ethical use in machine learning and paying attention to how your work is used
-    - https://github.com/EthicalML/awesome-artificial-intelligence-guidelines
-
-## How To Use This Book
-This book aims to provide an introduction to the commonly used algorithms used for machine learning and data science in the Rust ecosystem. Each chapter will have a description of an algorithm, and walk through a code example from start to finish.
-
-Each of those code examples are located in the `code/` directory, and can be run independently of the book. For example, to run the entirety of the code example for the KMeans algorithm, the steps would look like the following:
+## What is this book for?
+
+This book aims to provide an accessible introduction to doing machine learning and data science in the Rust ecosystem. Each chapter will have the description of an algorithm, and walk through a code example from start to finish.
+
+## Who is this book for?
+
+This book is written with two primary audiences in mind: developers who are familiar with machine learning and want to write their code Rust, and developers who are familiar with Rust and want to do some machine learning. 
+
+In both cases, we generally assume a basic level of understanding of the Rust programming language, although mastery is certainly not required! If you're brand new to the language, it's suggested to start off by reading [*The Rust Programming Language*](https://doc.rust-lang.org/book/), then returning when you feel a little more comfortable. Perhaps just as importantly as Rust's syntax, a familiarity with the library/crates ecosystem and documentation practices will prove very valuable. Machine learning in many cases sits near the top of the stack; especially when one is working with data, there are usually several layers of code beneath what the top one is doing. That's one of the benefits of working in Rust; these lower layers are often also written in Rust, which makes the abstraction more transparent and empowers developers to dig fearlessly into the underlying aspects of these programs.
+
+Conversely, we don't assume an in-depth knowledge of machine learning. Some familiarity with the algorithms may be helpful, but the descriptions and code contained in here should help to build a foundation of some of these topics.  
```

Comment:
the in-depth knowledge is a bit too vague for me, perhaps you can include math into it: `assume an in-depth knowledge  of machine learning (i.e. mathematical familiarity of the field)`